tornado/test/simple_httpclient_test.py

```diff
@@ -322,12 +330,11 @@ def test_options_request(self):
     def test_no_content(self):
         response = self.fetch("/no_content")
         self.assertEqual(response.code, 204)
-        # 204 status doesn't need a content-length, but tornado will
-        # add a zero content-length anyway.
+        # 204 status shouldn't have a content-length
         #
         # A test without a content-length header is included below
         # in HTTP204NoContentTestCase.
```

Comment:
There used to be two tests related to 204: This one in which the Content-Length header was used, and HTTP204NoContentTestCase, which manipulated the connection manually to create responses without Content-Length. Now HTTP204NoContentTestCase is redundant; it should be reversed and become the test of responses *with* the header.